src/useVisibilityObserver/useVisibilityObserver.ts

```diff
@@ -0,0 +1,29 @@
+import { onMounted, ref, Ref } from 'vue'
+import { useIntersectionObserver, UseIntersectionObserverOptions } from '@/useIntersectionObserver'
+
+export type UseVisibilityObserverResponse = {
+  visible: Ref<boolean>,
+  disconnect: () => void,
+}
+
+export function useVisibilityObserver(element: Ref<HTMLElement | undefined>, disconnectWhenVisible: boolean = false, options: UseIntersectionObserverOptions = {}): UseVisibilityObserverResponse {
```

Comment:
Can we combine `disconnectWhenVisible` with the other `options` so we accept only two arguments? `element` and `options`.